api/pkg/instrumentation/metrics/prometheus_test.go

```diff
@@ -10,6 +11,102 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+// mockGauge mocks a prometheus Gauge
+type mockGauge struct {
+	mock.Mock
+	count float64
+}
+
+func (g *mockGauge) Desc() *prometheus.Desc {
```

Comment:
Since these mock methods are implementing multiple interfaces, some comments to separate them would be useful. Eg:
```
// Implementing prometheus.Metric
func (g *mockGauge) Desc() *prometheus.Desc {
...
func (g *mockGauge) Write(*io_prometheus_client.Metric) error {
...

// Implementing prometheus.Collector
func (g *mockGauge) Describe(chan<- *prometheus.Desc) {
...

```